pymc4/distributions/continuous.py

```diff
@@ -1174,6 +1177,84 @@ def upper_limit(self):
         return self.conditions["high"]
 
 
+class Flat(ContinuousDistribution):
+    r"""A uniform distribution with support :math:`(-\inf, \inf)`.
+    Used as a uninformative log-likelihood that returns
+    zeros regardless of the passed values.
+    """
+
+    def __init__(self, name, **kwargs):
+        super().__init__(name, **kwargs)
+
+    @staticmethod
+    def _init_distribution(conditions):
+        return tfd.Uniform(low=-np.inf, high=np.inf)
+
+    def log_prob(self, value):
+        # convert the value to tensor
+        value = tf.convert_to_tensor(value)
+        expected = tf.zeros(self._distribution.batch_shape + self._distribution.event_shape)
+        # check if the event shape matches
+        if (
+            len(self._distribution.event_shape)
+            and value.shape[-len(self._distribution.event_shape) :]
+            != self._distribution.event_shape
+        ):
+            raise ValueError("values not consistent with the event shape of distribution")
+        # broadcast expected to shape of value
+        try:
+            expected = tf.broadcast_to(expected, value.shape)
```

Comment:
Does this work if `value.shape=(3,)` but the distribution has `batch_shape=(4,)` and `event_shape=(3,)`, so the expected shape is `(4, 3)`? Wont it raise an error for that kind of situation?

Reply:
Yes. I think this can be fixed by checking ``tf.rank(values) < tf.rank(self.batch_shape + self.event_shape)``. If this condition evaluates to ``True``,  we only check if the values are consistent with ``batch_shape`` and we don't need to broadcast. While if the condition is ``False``,  let ``tf.broadcast_to`` handle it. What do you say? Is there a way around?

<details>
<summary>Code</summary>

```python
if tf.rank(value) < tf.rank(self.batch_shape + self.event_shape):
    if value.shape[:-len(self.event_shape)] != self.batch_shape[-1:-(len(value.shape)-len(self.event_shape))]:
        raise ValueError("batch shape of values is not consistent with distribution's batch_shape")
else:
    try:
        ...
```

</details>